pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -646,6 +656,56 @@ func (r *controlPlaneInstanceReconciler) renderCharts(ctx context.Context) error
 	return nil
 }
 
+func (r *controlPlaneInstanceReconciler) validateManifests(ctx context.Context) error {
+	log := common.LogFromContext(ctx)
+	allErrors := []error{}
+	// validate resource namespaces
+	smmr := &v1.ServiceMeshMemberRoll{}
+	if err := r.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Instance.GetNamespace(), Name: common.MemberRollName}, smmr); err != nil {
+		if !apierrors.IsNotFound(err) {
+			// log error, but don't fail validation: we'll just assume that the control plane namespace is the only namespace for now
+			log.Error(err, "failed to retrieve SMMR for SMCP")
```

Comment:
It's probably worth returning the error and letting the reconciliation restart.  If there's a gateway that references a namespace that is legitimate, then we'll error when we validate, which will be a false error message.

Reply:
So you want to restart reconciliation without logging an error in this case?

Reply:
The thing is that this will probably only happen rarely, and even if it happens, it will only have an impact if you have a gateway outside the control plane namespace. The idea with just logging and continuing was that 99% of the time it'll be alright, because the gateway is in the cp namespace anyway

Reply:
I believe you can just return the error from the reconcile loop and let it do its thing.  I don't remember if we need to log it and return it, or just return it.  @luksa?

Reply:
I agree that it's probably unlikely that it will fail and unlikely that a namespace other than the control plane will be used.  Maybe keep track of the fact that the smmr lookup failed and then _if_ there's a validation failure, you can return the smmr error instead. wdyt?

Reply:
Yeah, I think that makes sense

Reply:
I fixed it now. While at it, I found another issue; I had forgotten to split the manifests (by "---") so was only looking at the first manifest in the output of each helm template (which for the gateways are multiple).